.eslintrc.json

```diff
@@ -75,6 +75,7 @@
     },
     {
       "files": [
+        "assets/js/components/*.js",
```

Comment:
I added `"assets/js/components/**/*.js",` to the list of files to lint and got only 44 errors, all of which are expected (it intelligently lints `index.js` as-expected.)

Is there another issue for dealing with components in folders? If so and we want to wait until that one then okay, but otherwise it makes sense to me for this to include folders (eg `assets/js/components/legacy-setup/site-verification.js` should be `assets/js/components/legacy-setup/SiteVerification.js`, etc.)

Reply:
Hey Matt, that covers a lot more cases, I'll update and review the new linter issues that come up.

Reply:
Also, I have to make sure the imports are updated to the new name as well.

Reply:
Hey @tofumatt,

Working through one of the issues I can see is:

```
/google-site-kit/assets/js/components/data/index.js
  22:1  error  The directory 'data' must be named 'dataAPI', after the exported value of its index file  filenames/match-exported
```
Do we want to rename the `data` directory to `dataAPI`?

Also, there are a lot of changes in the `legacy-` prefixed folders. Is it worth adding an exclusion to these as I'm assuming they're planning to be phased out over time so it would be unnecessary to rename and bulk update imports for them?

Reply:
Ah, I see. Let's not rename `data`, it's arguably legacy too and best to leave alone (it likely has a lot of imports).

Excluding the `legacy-` folders definitely makes sense, exactly as you say. Let's exclude them and the `data` folder, but everything else makes sense to update. 👍🏻

Reply:
In the end, I went in and worked through as many files as I could. git is a bit finicky with changing the case of existing file and folder names on a case insensitive filesystem like on the mac so I had a lot of trouble committing the code and husky also cased issues killing the commit. Can you pull down this branch and see if all of the changes happen correctly?